src/main/python/ttconv/imsc/utils.py

```diff
@@ -114,22 +111,55 @@ def parse_length(attr_value: str) -> typing.Tuple[float, str]:
   raise ValueError("Bad length syntax")
 
 
+_FAMILIES_ESCAPED_CHAR = re.compile(r"\\(.)")
+_SINGLE_QUOTE_PATTERN = "(?:'(?P<single_quote>(.+?)(?<!\\\\))')"
+_DOUBLE_QUOTE_PATTERN = "(?:\"(?P<double_quote>(.+?)(?<!\\\\))\")"
+_NO_QUOTE_PATTERN = "(?P<no_quote>(?:\\\\.|[^'\", ])(?:\\\\.|[^'\",])+)"
+
+_FONT_FAMILY_PATTERN = re.compile(
+  "|".join(
+    (
+      _SINGLE_QUOTE_PATTERN,
+      _DOUBLE_QUOTE_PATTERN,
+      _NO_QUOTE_PATTERN
+    )
+  )
+)
+
 def parse_font_families(attr_value: str) -> typing.List[str]:
   '''Parses th TTML \\<font-family\\> value in `attr_value` into a list of font families'''
 
   rslt = []
 
-  for family in map(str.strip, _FAMILIES_SEPARATOR.split(attr_value)):
+  for m in _FONT_FAMILY_PATTERN.finditer(attr_value):
 
-    unquoted_family = family[1:-1] if family[0] == "'" or family[0] == '"' else family
+    is_quoted = m.lastgroup in ("single_quote", "double_quote")
 
-    rslt.append(_FAMILIES_ESCAPED_CHAR.sub(r"\1", unquoted_family))
+    escaped_family = _FAMILIES_ESCAPED_CHAR.sub(r"\1", m.group(m.lastgroup))
+
+    if not is_quoted and escaped_family in styles.GenericFontFamilyType.__members__:
+      rslt.append(styles.GenericFontFamilyType(escaped_family))
+    else:
+      rslt.append(escaped_family)
+    
 
   if len(rslt) == 0:
     raise ValueError("Bad syntax")
 
   return rslt
 
+def serialize_font_family(font_family: typing.Tuple[typing.Union[str, styles.GenericFontFamilyType]]):
```

Comment:
This typing hint should define the second type of the tuple, shouldn't it?

Reply:
Fixed at https://github.com/sandflow/ttconv/pull/195/commits/deacad006a524b157800075d3e5c0cc3a18630c1